Add Tini/docker-init documentation (#24540)
docker/docker.github.io: content/manuals/engine/containers/multi-service_container.md

full-fledged init process such as `sysvinit` or `systemd` to handle process
lifecycle within your container.

## Use the `--init` flag

Docker Engine ships with [Tini](https://github.com/krallin/tini), a minimal
init process built specifically for containers. The bundled binary is called
`docker-init`. When you pass `--init` to `docker run`, the daemon mounts
`docker-init` into the container and runs it as PID 1. It performs two
functions:

> - Forwards signals (such as `SIGTERM`) to child processes so they can shut
>   down gracefully

Reviewer: Tini forwards signals to its immediate child by default, rather than to every descendant process. In this setup, that child is the container's main process.

```suggestion
- Forwards signals (such as `SIGTERM`) to the container's main process so it
  can shut down gracefully
```

Generated by Codex

- Reaps zombie processes that would otherwise accumulate when child processes
  exit

Run a container with `--init` and inspect its process tree:

```console
$ docker run -dit --init busybox
a428c601d073...

$ docker top a428c601d073
UID   PID     PPID    C   STIME   TTY   TIME       CMD
root  25165   25145   0   12:49   ?     00:00:00   /sbin/docker-init -- sh
root  25189   25165   0   12:49   ?     00:00:00   sh
```

> The output shows `/sbin/docker-init` running as PID 1, with the shell as its
> child process.

Reviewer: The preceding text already explains PID 1 and reaping. Keep this interpretation focused on what the process tree adds:

```suggestion
`/sbin/docker-init` runs as the shell's parent process.
```

Generated by Codex
